library/objective-c/EnvoyEngineImpl.m

```diff
@@ -66,40 +62,51 @@ static envoy_headers toNativeHeaders(EnvoyHeaders *headers) {
   return (envoy_headers){length, header_array};
 }
 
+static const void* ios_http_filter_init(const void *context) {
+  EnvoyHTTPFilterFactory *filterFactory = (__bridge EnvoyHTTPFilterFactory *)context;
+  EnvoyHTTPFilter *filter = filterFactory.create();
+  return CFBridgingRetain(filter);
+}
+
 static envoy_filter_headers_status
-ios_http_filter_on_request_headers(envoy_headers headers, bool end_stream, void *context) {
+ios_http_filter_on_request_headers(envoy_headers headers, bool end_stream, const void *context) {
   // TODO(goaway): optimize unmodified case
-  ios_http_filter_context *c = (ios_http_filter_context *)context;
-  if (c->filter.onRequestHeaders == nil) {
+  EnvoyHTTPFilter *filter = (__bridge EnvoyHTTPFilter *)context;
```

Comment:
what does the `__bridge` in the cast mean?

Reply:
`__bridge` tells ARC that we're pulling out a retainable object here, but not to actually do anything with the retain count at this point.